grpc.go

```diff
@@ -993,14 +993,34 @@ func (a *agentGRPC) RemoveContainer(ctx context.Context, req *pb.RemoveContainer
 		if err := ctr.removeContainer(); err != nil {
 			return emptyResp, err
 		}
+
+		// Find the sandbox storage used by this container
+		for _, path := range ctr.mounts {
+			if _, ok := a.sandbox.storages[path]; ok {
+				if err := a.sandbox.unsetAndRemoveSandboxStorage(path); err != nil {
+					return emptyResp, err
+				}
+			}
+		}
 	} else {
 		done := make(chan error)
 		go func() {
 			if err := ctr.removeContainer(); err != nil {
 				done <- err
+				close(done)
+				return
 			}
 
-			close(done)
```

Comment:
This is still necessary, and should be moved to last line of this goroutine function.

Because if you don't close the done channel, the 
```
select {
 		case err := <-done:
 			if err != nil {
 				return emptyResp, err
 			}
 		case <-time.After(time.Duration(req.Timeout) * time.Second):
 			return emptyResp, grpcStatus.Errorf(codes.DeadlineExceeded, "Timeout reached after %ds", timeout)
 		}
 	}
```

`<-done` will always block if no error happens, and this function will always time out